store/rootmulti/store.go

```diff
@@ -386,15 +399,8 @@ func (rs *Store) Commit() types.CommitID {
 		version = previousHeight + 1
 	}
 
-	rs.lastCommitInfo = rs.commitStores(version, rs.stores)
-
-	var pruneErr error
-	defer func ()  {
-		flushMetadata(rs.db, version, rs.lastCommitInfo, rs.pruneHeights)
-		if pruneErr != nil {
-			panic(pruneErr)
-		}
-	}()
+	newCommitInfo := rs.commitStores(version, rs.stores)
+	rs.updateLatestCommitInfo(newCommitInfo, version)
```

Comment:
Updates `rs.lastCommitInfo` and flushes to disk atomically and synchronously with other reads of `rs.lastCommitInfo`